lib/relation-definition.js

```diff
@@ -107,6 +107,15 @@ function bindRelationMethods(relation, relationMethod, definition) {
   });
 };
 
+function preventFkOverride(inst, data, fkProp) {
+  var err;
+  if (!fkProp) return err;
```

Comment:
I think we need to create a new Error then return it? Or maybe I missed some code that creates it?

Reply:
It returns `undefined`, the location where the function is called checks to see if err is `undefined` or not and passes the error to the callback and returns to stop execution.